src/foundry/foundry.js/collections/documentCollections/compendiumCollection.d.ts

```diff
@@ -0,0 +1,206 @@
+     */
+    deleteCompendium(): Promise<this>;
+
+    /**
+     * Duplicate a compendium pack to the current World.
+     * @param label - A new Compendium label
+     */
+    duplicateCompendium({ label }: { label?: string }): Promise<this>;
+
+    /**
+     * Validate that the current user is able to modify content of this Compendium pack
+     * @remarks `requireUnlocked` has a default value of `true`
+     */
+    protected _assertUserCanModify({ requireUnlocked }: { requireUnlocked?: boolean }): boolean;
+
+    /** Request that a Compendium pack be migrated to the latest System data template */
+    migrate(options?: object): Promise<this>;
+
+    _onCreateDocuments(
+      documents: DocumentInstanceForMetadata<T>[],
+      result: DocumentInstanceForMetadata<T>['data']['_source'][],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    _onUpdateDocuments(
+      documents: DocumentInstanceForMetadata<T>[],
+      result: DeepPartial<DocumentInstanceForMetadata<T>>[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    _onDeleteDocuments(
+      documents: DocumentInstanceForMetadata<T>[],
+      result: string[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    /**
+     * Follow-up actions taken when Documents within this Compendium pack are modified
+     */
+    protected _onModifyContents(
+      documents: DocumentInstanceForMetadata<T>[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    /** @deprecated since 0.8.0 */
+    get entity(): this['documentClass']['documentName'];
+
+    /** @deprecated since 0.8.0 */
+    get content(): ReturnType<this['getDocuments']>;
+
+    /** @deprecated since 0.8.0 */
+    getEntry(id: string): Promise<DocumentInstanceForMetadata<T>['data']>;
+
+    /** @deprecated since 0.8.0 */
+    getEntity(id: string): ReturnType<this['getDocument']>;
+
+    /** @deprecated since 0.8.0 */
+    importEntity(document: DocumentInstanceForMetadata<T>): ReturnType<this['importDocument']>;
+
+    /** @deprecated since 0.8.0 */
+    createEntity(data: any, options: Partial<DocumentModificationOptions>): ReturnType<this['documentClass']['create']>;
+
+    /** @deprecated since 0.8.0 */
+    updateEntity(data: any, options: Partial<DocumentModificationOptions>): DocumentInstanceForMetadata<T>['update'];
+
+    /** @deprecated since 0.8.0 */
+    deleteEntity(id: string, options: Partial<DocumentModificationOptions>): DocumentInstanceForMetadata<T>['delete'];
+  }
+
+  interface CompendiumConfiguration {
+    private: boolean;
+    locked: boolean;
+  }
+
+  interface ImportAllOptions {
+    /** An existing Folder _id to use. */
+    folderId: string;
+    /** A new Folder name to create. */
+    folderName: string;
+    /** Additional options forwarded to Document.createDocuments */
+    options: object;
```

Comment:
We should type this better. I would guess it is either `DocumentModificationContext` or `DocumentModificationOptions`, but please have a look.

Reply:
It should be `DocumentModificationContext`